src/App.tsx

```diff
@@ -1,61 +1,117 @@
+};
 
 export const App = () => {
+  const [inputTitle, setInputTitle] = useState('');
+  const [selectUser, setSelectUser] = useState(0);
+  const [hasErrorMessageTitle, setHasErrorMassageTitle] = useState(false);
+  const [hasErrorMessageSelect, setHasErrorMassageSelect] = useState(false);
+  const [todosList, setTodosList] = useState(todosFromServer);
+
+  const handleTitleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputTitle(e.target.value);
+    setHasErrorMassageTitle(false);
+  };
+
+  const handleChangeSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectUser(+e.target.value);
+    setHasErrorMassageSelect(false);
+  };
+
+  const reset = () => {
+    setInputTitle('');
+    setSelectUser(0);
+    setHasErrorMassageTitle(false);
+    setHasErrorMassageSelect(false);
+  };
+
+  const addNewTodo = (newTodo: Todo) => {
+    setTodosList(prevList => [...prevList, newTodo]);
+  };
+
+  const addTodo = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+
+    setHasErrorMassageTitle(!inputTitle);
+    setHasErrorMassageSelect(!selectUser);
+
+    if (!inputTitle || !selectUser) {
+      return;
+    }
+
+    addNewTodo({
+      id: getMaxId(),
+      title: inputTitle,
+      userId: selectUser,
+      completed: false,
+    });
+    reset();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
       <form action="/api/todos" method="POST">
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">Title:&nbsp;</label>
+          <input
+            type="text"
+            value={inputTitle}
+            data-cy="titleInput"
+            onChange={handleTitleChange}
+            required
+            placeholder="Enter a tittle"
+            id="titleInput"
+          />
+          {hasErrorMessageTitle && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
+          <label htmlFor="userSelect">User:&nbsp;</label>
+          <select
+            data-cy="userSelect"
+            value={selectUser}
+            onChange={handleChangeSelect}
+            required
+            id="userSelect"
+          >
             <option value="0" disabled>
               Choose a user
             </option>
+            {usersFromServer.map(user => (
+              <option value={user.id} key={user.id}>
+                {user.name}
+              </option>
+            ))}
           </select>
-
-          <span className="error">Please choose a user</span>
+          {hasErrorMessageSelect && (
+            <span className="error">Please choose a user</span>
+          )}
         </div>
 
-        <button type="submit" data-cy="submitButton">
+        <button type="submit" data-cy="submitButton" onClick={addTodo}>
```

Comment:
[CODE KNOWLEDGE] - While handling form values, trigger submit logic using `onSubmit` prop of form, instead of `onClick` submit button. So, you need to remove onClick prop from here.